src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,292 @@
+     * @param sc Scanner object to register user's input
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> toDo(Scanner sc, TaskList<Task> tasks) {
+        String description = sc.nextLine();
+        if (description.trim().length() == 0) {
+            throw new DukeException("Todo must not be empty");
+        }
+        Task newTask = new Todos(description);
+        return tasks.add(newTask);
+    }
+
+    /**
+     * Add a new Deadline task into the list of tasks from the saved data.
+     * @param description Title of the task
+     * @param date End date of deadline task
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> deadline(String description, String date, TaskList<Task> tasks) {
+        if (description.trim().length() == 0) {
+            throw new DukeException("Deadline must not be empty");
+        }
+        try {
+            Task newTask = new Deadline(description, date);
+            return tasks.add(newTask);
+        } catch (DukeException e) {
+            System.out.println("Todo must not be empty");
+        }
+        return tasks;
+    }
+    /**
+     * Add a new Deadline task into the list of tasks from the user's input.
+     * @param sc Scanner object to register user's input
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> deadline(Scanner sc, TaskList<Task> tasks) {
+        String description = sc.nextLine();
+        if (description.trim().length() == 0) {
+            throw new DukeException("Deadline must not be empty");
+        }
+        try {
+            String[] dateRange = description.split("/by");
+            Task newTask = new Deadline(dateRange[0], dateRange[1].replaceFirst("by", ""));
+            return tasks.add(newTask);
+        } catch (DukeException e) {
+            System.out.println("Todo must not be empty");
+        }
+        return tasks;
+    }
+    /**
+     * Add a new Event task into the list of tasks from the saved data.
+     * @param description Title of the task
+     * @param from Start date of the event task
+     * @param end Stipulated last date of the event task
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> events(String description, String from, String to, TaskList<Task> tasks) {
```

Comment:
More intuitive name here would be nice too! 
```suggestion
    static TaskList<Task> addEvent(String description, String from, String to, TaskList<Task> tasks) {
```